### PR TITLE
feat: add accent to node props for treepicker

### LINKED ISSUE #980: TreePicker node requires dynamically changing color base on `node` data.
## Types of issue

- [ ] Bug (High impact or critical to a production build)
- [ ] Bug (Low impact or not working as expected)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Something else (idea, build tool, documentation improvement)

## Description

New design requires `TreePicker` node requires dynamically changing background color base on `node` data.

Some WIP proposal PR in https://github.com/Adslot/adslot-ui/pull/979, by injecting classname dynamically.

### PR DESCRIPTION
**Pull request by xiaofan2406**
## Description

- add `accent` to support `warning` or `error`
- add lighten versions of the current colors 

Fixes #980 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/components/TreePicker/Grid/index.jsx [hunk 1]
```diff
@@ -83,6 +85,7 @@ TreePickerGridComponent.propTypes = {
   isLoading: PropTypes.bool,
   nodes: PropTypes.arrayOf(TreePickerPropTypes.node),
   nodeRenderer: PropTypes.func,
+  rowClassNames: PropTypes.func,
```
Comment (xiaofan2406) [inline src/components/TreePicker/Grid/index.jsx]: Naming very much open to suggestion

**Comment by veegandhi** [comment]
Per offline discussion. This needs a issue card on the repo.

**Comment by veegandhi** [comment]
Can you add accent to one of the node selections in the docs?

File: src/components/TreePicker/Node/styles.scss [hunk 2]
```diff
@@ -17,6 +17,14 @@
     width: $icon-size;
   }
 
+  &.is-error {
```
Comment (veegandhi) [inline src/components/TreePicker/Node/styles.scss]: Also needs success.

File: src/components/TreePicker/Node/index.jsx [hunk 3]
```diff
@@ -54,8 +55,12 @@ class TreePickerNodeComponent extends React.PureComponent {
 
     const labelCellProps = isExpandable && !this.state.isLoading ? { onClick: this.setLoadingAndExpandNode } : {};
 
+    const classNames = classnames(baseClass, isChildNode && 'child-node', {
+      [`is-${node.accent}`]: node.accent,
+    });
```
Comment (lightbringer1991) [inline src/components/TreePicker/Node/index.jsx]: I guess we can follow the classnames convention here
```
const classNames = classnames(baseClass, {
  'child-node': isChildNode,
  [`is-${node.accent}`]: node.accent,
});
```

File: src/prop-types/TreePickerPropTypes.js [hunk 4]
```diff
@@ -17,6 +17,7 @@ export default {
       ancestors: PropTypes.arrayOf(PropTypes.shape(baseNodeProps).isRequired),
       type: PropTypes.string.isRequired,
       value: PropTypes.number,
+      accent: PropTypes.string,
```
Comment (lightbringer1991) [inline src/prop-types/TreePickerPropTypes.js]: I guess a list of valid accents would be great

**Pull request merged**